src/prelude/internals.mo

```diff
@@ -444,8 +446,13 @@ func @install_actor_helper(
         (#reinstall, (prim "cast" : (actor {}) -> Principal) actor1)
       };
       case (#upgrade actor2) {
+        let wasm_memory_persistence = if enhanced_orthogonal_persistence { 
+          ?(#Keep) 
```

Comment:
What will happen here if the old actor doesn't support eop. Will it fail or just degrade to candid destabilization?

I'm thinking about the distributed map example. If we upgrade the main node from non-eop to eop, and it tries to upgrade it previously deployed non-eop actor class instances, will they fail to upgrade?

Reply:
This will fall back to upgrade by Candid stabilization, meaning the old actor is automatically migrated to EOP.
I added a test case `map-mixed-upgrades.drun` that tests the various scenarios of keeping and dropping main memory (including graph copy).